src/f32-gemm/MrxLmulv-rvv.c.in

```diff
@@ -0,0 +1,170 @@
+          v_acc${M} = __riscv_vfmacc_vf_f32m${LMUL}(v_acc${M}, *a${M}, v_w_k, vl);
+          a${M}++;
+        k -= sizeof(float);
+      } while (k != 0);
+      $if ACTIVATION == "MINMAX":
+        // clamp results with min & max
+        $for M in range(MR):
+          v_acc${M} = __riscv_vfmax_vf_f32m${LMUL}(v_acc${M}, vmin, vl);
+
+        $for M in range(MR):
+          v_acc${M} = __riscv_vfmin_vf_f32m${LMUL}(v_acc${M}, vmax, vl);
+      $elif ACTIVATION == "RELU":
+        // apply ReLU to results
+        $for M in range(MR):
+          v_acc${M} = __riscv_vfmax_vf_f32m${LMUL}(v_acc${M}, 0.0f, vl);
+      // store ${MR} x vl results to c
+      $for M in range(MR):  
+        __riscv_vse32_v_f32m${LMUL}(c${M}, v_acc${M}, vl);
+        c${M} = (float*) ((uintptr_t) c${M} + cn_stride);
+      $for M in range(MR):  
+        a${M} = (const float*) ((uintptr_t) a${M} - kc);
+    } while (nc != 0);
+    return;
+  }
+  $if MR > 1:
+    do {
+      if XNN_UNLIKELY(nc < nr) {
+        vl = __riscv_vsetvl_e32m${LMUL}(nc);
+      }
+      nc = nc - vl;
+
+      vfloat32m${LMUL}_t v_acc0 =  __riscv_vle32_v_f32m${LMUL}(w, vl);
+      w = w + nr;
+      $for M in range(1, MR-1):
+        vfloat32m${LMUL}_t v_acc${M};
+      switch (mr) {
+        $for M in range(MR-1, 1, -1):
+          case ${M}:
+            v_acc${M-1} =  __riscv_vmv_v_v_f32m${LMUL}(v_acc0, vl);
+        default:
+          break;
+      }
+      
+      size_t k = kc;
+      do {
+        vfloat32m${LMUL}_t v_w_k = __riscv_vle32_v_f32m${LMUL}(w, vl);
+        w = w + nr;
+        switch (mr) {
+          $for M in range(MR-1, 1, -1):
+            case ${M}:
+              v_acc${M-1} = __riscv_vfmacc_vf_f32m${LMUL}(v_acc${M-1}, *a${M-1}, v_w_k, vl);
+              a${M-1}++;
+          default:
+            v_acc0 = __riscv_vfmacc_vf_f32m${LMUL}(v_acc0, *a0, v_w_k, vl);
+            a0++;
+            break;
+        }
+        k -= sizeof(float);
+      } while (k != 0);
+      $if ACTIVATION == "MINMAX":
+        switch (mr) {
+          $for M in range(MR-1, 1, -1):
+            case ${M}:
+              v_acc${M-1} = __riscv_vfmax_vf_f32m${LMUL}(v_acc${M-1}, vmin, vl);
+          default:
+            v_acc0 = __riscv_vfmax_vf_f32m${LMUL}(v_acc0, vmin, vl);
+            break;
+        }
+        switch (mr) {
+          $for M in range(MR-1, 1, -1):
+            case ${M}:
+              v_acc${M-1} = __riscv_vfmin_vf_f32m${LMUL}(v_acc${M-1}, vmax, vl);
+          default:
+            v_acc0 = __riscv_vfmin_vf_f32m${LMUL}(v_acc0, vmax, vl);
+            break;
+        }
+      $elif ACTIVATION == "RELU":
+        switch (mr) {
+          $for M in range(MR-1, 1, -1):
+            case ${M}:
+              v_acc${M-1} = __riscv_vfmax_vf_f32m${LMUL}(v_acc${M-1}, 0.0f, vl);
+          default:
+            v_acc0 = __riscv_vfmax_vf_f32m${LMUL}(v_acc0, 0.0f, vl);
+            break;
+        }
+      switch (mr) {
+        $for M in range(MR-1, 1, -1):
```

Comment:
stores for GEMM can be forwards.  IGEMM and GEMMINC are backwards, but dont worry about those.
but you'll need to clamp a and c pointers and compute redundent accumulators.

Reply:
> you'll need to clamp a and c pointers and compute redundent accumulators.

Yes, this is why I tackle `mr < MR` case behind if statement.

Reply:
Change to follow other platforms:
```
  $for M in range(1, MR):
    const float* a${M} = (const float*) ((uintptr_t) a${M-1} + a_stride);
    float* c${M} = (float*) ((uintptr_t) c${M-1} + cm_stride);
    $if M % 2 == 0:
      if XNN_UNPREDICTABLE(mr <= ${M}) {
        a${M} = a${M-1};
        c${M} = c${M-1};
      }
    $elif M + 1 == MR:
      if XNN_UNPREDICTABLE(mr != ${M+1}) {
        a${M} = a${M-1};
        c${M} = c${M-1};
      }
    $else:
      if XNN_UNPREDICTABLE(mr < ${M+1}) {
        a${M} = a${M-1};
        c${M} = c${M-1};
      }
```

Reply:
Does the current method run faster than switch cases method which I provided?
If `mr < target MR`, we still need to redundant store and computation.

```
  $for M in range(1, MR):
    const float* a${M} = (const float*) ((uintptr_t) a${M-1} + a_stride);
    float* c${M} = (float*) ((uintptr_t) c${M-1} + cm_stride);
    $if M % 2 == 0:
      if XNN_UNPREDICTABLE(mr <= ${M}) {
        a${M} = a${M-1};
        c${M} = c${M-1};
      }
    $elif M + 1 == MR:
      if XNN_UNPREDICTABLE(mr != ${M+1}) {
        a${M} = a${M-1};
        c${M} = c${M-1};
      }
    $else:
      if XNN_UNPREDICTABLE(mr < ${M+1}) {
        a${M} = a${M-1};
        c${M} = c${M-1};
      }
```